Options.h

```diff
@@ -0,0 +1,199 @@
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+// Description: A class defining the parameters to the BasisGenerator class and
+// SVD algorithm.
+
+#ifndef included_Options_h
+#define included_Options_h
+
+namespace CAROM {
+
+/**
+ * A class defining the parameters to the BasisGenerator class and SVD algorithm.
+ */
+class Options
+{
+  /**
+   * @brief Constructor.
+   *
+   * @pre dim > 0
+   * @pre samples_per_time_interval > 0
+   * @pre max_basis_dimension > 0
+   * @pre singular_value_tol >= 0
+   *
+   * @param[in] dim The dimension of the system on this processor.
+   * @param[in] samples_per_time_interval The maximum number of samples in
+   *                                      each time interval.
+   * @param[in] max_time_intervals The maximum number of time intervals.
+   * @param[in] update_rightSV Whether to update the right SV or not.
+   * @param[in] write_snapshots Whether to automatically write snapshots matrices
+   *                        instead of basis matrices.
+   * @param[in] maximum basis dimension (typemax(int)) The maximum number of
+   *                                vectors returned in the basis.
+   * @param[in] singular_value_tol Tolerance to determine whether to include
+   *                               a singular value in the SVD.
+   * @param[in] debug_algorithm If true results of static Options algorithm
+   *                            will be printed to facilitate debugging.
+   *
+   * Static SVD
+   *
+   * If both max_basis_dimension and singular_value_tol would result in
+   * truncating the basis, the dimension of the returned basis will be the
+   * *minimum* of the number of vectors that is computed from each.
+   *
+   *
+   * Incremental SVD
+   *
+   * @pre linearity_tol > 0.0
+   * @pre max_basis_dimension <= dim
+   * @pre initial_dt > 0.0
+   * @pre sampling_tol > 0.0
+   * @pre max_time_between_samples > 0.0
+   * @pre min_sampling_time_step_scale >= 0.0
+   * @pre sampling_time_step_scale >= 0.0
+   * @pre max_sampling_time_step_scale >= 0.0
+   * @pre min_sampling_time_step_scale <= max_sampling_time_step_scale
+   *
+   * @param[in] linearity_tol Tolerance to determine whether or not a
+   *                          sample is linearly dependent.
+   * @param[in] skip_linearly_dependent If true skip linearly dependent
+   *                                    samples.
+   * @param[in] fast_update If true use the fast update algorithm.
+   * @param[in] initial_dt Initial simulation time step.
+   * @param[in] sampling_tol Sampling control tolerance.  Limits error in
+   *                         projection of solution into reduced order
+   *                         space.
+   * @param[in] max_time_between_samples Upper bound on time between
```

Comment:
Since these options are public, it would be helpful if they were listed in the order they appear in the functions

Reply:
I have updated the comments